server/src/main/java/org/opensearch/http/AbstractHttpServerTransport.java

```diff
@@ -352,21 +363,63 @@ protected void serverAcceptedChannel(HttpChannel httpChannel) {
      * @param httpChannel that received the http request
      */
     public void incomingRequest(final HttpRequest httpRequest, final HttpChannel httpChannel) {
-        handleIncomingRequest(httpRequest, httpChannel, httpRequest.getInboundException());
+        try (ThreadContext.StoredContext ignore = threadPool.getThreadContext().stashContext()) {
+            // TODO: Add support for parsing the otel incoming tracer.
+            final SpanScope httpRequestSpanScope = tracer.startSpan(createSpanName(httpRequest), buildSpanAttributes(httpRequest));
+            httpRequestSpanScope.addSpanAttribute(AttributeNames.HTTP_REQ_INBOUND_EX, httpRequest.getInboundException() != null);
+            HttpChannel traceableHttpChannel = new TraceableHttpChannel(httpChannel, httpRequestSpanScope);
+            handleIncomingRequest(httpRequest, traceableHttpChannel, httpRequest.getInboundException());
+        }
+    }
+
+    private String createSpanName(HttpRequest httpRequest) {
+        return httpRequest.method().name() + " " + httpRequest.uri();
+    }
+
+    private Attributes buildSpanAttributes(HttpRequest httpRequest) {
+        Attributes attributes = Attributes.create()
+            .addAttribute(AttributeNames.HTTP_URI, httpRequest.uri())
+            .addAttribute(AttributeNames.HTTP_METHOD, httpRequest.method().name())
+            .addAttribute(AttributeNames.HTTP_PROTOCOL_VERSION, httpRequest.protocolVersion().name());
+        populateHeader(httpRequest, attributes);
+        return attributes;
+    }
+
+    private void populateHeader(HttpRequest httpRequest, Attributes attributes) {
+        HEADERS_TO_BE_POPULATED.forEach(x -> {
+            if (httpRequest.getHeaders() != null && httpRequest.getHeaders().get(x) != null) {
+                attributes.addAttribute(x, Strings.collectionToCommaDelimitedString(httpRequest.getHeaders().get(x)));
+            }
+        });
     }
 
     // Visible for testing
     void dispatchRequest(final RestRequest restRequest, final RestChannel channel, final Throwable badRequestCause) {
+        final SpanScope spanScope = tracer.startSpan(createRestRequestSpanName(restRequest));
```

Comment:
```suggestion
        final SpanScope spanScope = tracer.startSpan(createRestRequestSpanName(restRequest.method() + " " + restRequest.path()));
```

Reply:
method and path will be almost same coming from HttpRequest, so doing this - restRequest.method() + " " + restRequest.path() + request.getRequestId()